lib.nix

```diff
@@ -125,7 +114,7 @@ let
     yarnRcYml ? src + "/.yarnrc.yml",
     yarnPlugins ? src + "/.yarn/plugins",
     yarnPath ? getYarnPath src,
-    yarn ? lib.mapNullable mkYarnBin { inherit yarnPath; },
+    yarn ? wrapYarnWith { yarn = yarnPath; },
```

Comment:
I don't think the mapNullable had any effect here, since the attribute set argument was always != null.

Reply:
`getYarnPath` is possibly null if it couldn't find a yarn release. You'd then eventually get an error that `yarn` wasn't provided.

Reply:
Wouldn't that have to be
```nix
{
  yarn ? if yarnPath != null then mkYarnBin { inherit yarnPath; } else yarnPath,
}
```
or with `mapNullable` 
```nix
{
  yarn ? lib.mapNullable (yarnPath: mkYarnBin { inherit yarnPath; }) yarnPath,
}
```
?

Reply:
🤦 I think I initially made `mkYarnBin` accept just `yarnPath` and it was correct then and then blindly made in an attrset.